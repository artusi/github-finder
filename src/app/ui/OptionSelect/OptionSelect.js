import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./OptionSelect.module.css";

export class OptionSelect extends Component {
  constructor(props) {
    super(props);
    this.state = { value: props.initialValue };

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.setState({ value: event.target.value });
    this.props.onChange(event.target.value);
  }

  render() {
    const { id, customClassName, disabled, options } = this.props;

    return (
      <select
        id={id}
        className={`${styles.optionSelect} ${customClassName}`}
        disabled={disabled}
        onChange={this.handleChange}
        value={this.state.value}
      >
        {options.map(item => (
          <option value={item.value} key={item.value}>
            {item.label}
          </option>
        ))}
      </select>
    );
  }
}

OptionSelect.propTypes = {
  id: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  options: PropTypes.array.isRequired,
  customClassName: PropTypes.string,
  initialValue: PropTypes.string,
  disabled: PropTypes.bool
};

OptionSelect.defaultProps = {
  customClassName: "",
  initialValue: "",
  disabled: false
};

export default OptionSelect;
