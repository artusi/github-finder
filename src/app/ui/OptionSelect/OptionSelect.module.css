.submitButton {
  font-weight: bold;
  padding: 6px 10px;
  background: #00324d;
  color: #fff;
  border: 2px solid #00324d;
  cursor: pointer;
}

.submitButton:hover {
  background: #fff;
  color: #00324d;
}

.submitButton:disabled {
  background: #5b6d77;
  border-color: #5b6d77;
  opacity: 0.8;
  cursor: not-allowed;
}

.submitButton:disabled:hover {
  background: #5b6d77;
  color: #fff;
}
