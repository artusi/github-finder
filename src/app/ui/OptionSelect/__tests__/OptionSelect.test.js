import React from "react";
import { OptionSelect } from "../OptionSelect";

const initialProps = {
  id: "sortRepos",
  options: [
    { value: "name", label: "name" },
    { value: "updated", label: "updated" },
    { value: "stars", label: "stars" }
  ],
  customClassName: "",
  initialValue: "",
  disabled: false,
  onChange: jest.fn()
};

describe("<OptionSelect /> - render", () => {
  it("renders", () => {
    const wrapper = shallow(<OptionSelect {...initialProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders with all props > customClassName | disabled", () => {
    const allProps = {
      ...initialProps,
      customClassName: "custom",
      initialValue: "updated",
      disabled: true
    };
    const wrapper = shallow(<OptionSelect {...allProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
