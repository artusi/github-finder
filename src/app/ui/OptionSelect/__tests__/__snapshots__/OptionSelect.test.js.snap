// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptionSelect /> - render renders 1`] = `
<select
  className="optionSelect "
  disabled={false}
  id="sortRepos"
  onChange={[Function]}
  value=""
>
  <option
    key="name"
    value="name"
  >
    name
  </option>
  <option
    key="updated"
    value="updated"
  >
    updated
  </option>
  <option
    key="stars"
    value="stars"
  >
    stars
  </option>
</select>
`;

exports[`<OptionSelect /> - render renders with all props > customClassName | disabled 1`] = `
<select
  className="optionSelect custom"
  disabled={true}
  id="sortRepos"
  onChange={[Function]}
  value="updated"
>
  <option
    key="name"
    value="name"
  >
    name
  </option>
  <option
    key="updated"
    value="updated"
  >
    updated
  </option>
  <option
    key="stars"
    value="stars"
  >
    stars
  </option>
</select>
`;
